fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/DefaultFilter.java

```diff
@@ -79,10 +78,7 @@ public DefaultFilter() {
     @Override
     public boolean apply(final Event event) {
         try {
-            final org.modeshape.jcr.api.observation.Event modeEvent = getJcr21Event(event);
-
-            final List<NodeType> nodeTypes = ImmutableList.copyOf(modeEvent.getMixinNodeTypes());
-            final Collection<String> mixinTypes = ImmutableSet.copyOf(transform(nodeTypes, nodetype2string));
+            final Collection<String> mixinTypes = getMixinTypes(event);
             return mixinTypes.contains(FEDORA_RESOURCE)
```

Comment:
Maybe put these three types in a collection and use `Collections.disjoint()` as below.